opera/rules.go

```diff
@@ -433,3 +434,59 @@ func (r Rules) String() string {
 	b, _ := json.Marshal(&r)
 	return string(b)
 }
+
+// FeatureSet is an enumeration of different releases, each one enabling a
+// different set of features. These are an abstraction that allows to reason
+// about the different releases instead of isolated upgrades.
+// Feature sets are exclusive, and a (fake-)net cannot be configured with
+// more than one feature set at the time.
+type FeatureSet int
+
+const (
+	SonicFeatures   FeatureSet = iota // < enables the initial sonic release features
+	AllegroFeatures                   // < enables the allegro release features
+)
+
+func (fs FeatureSet) ToUpgrades() (Upgrades, error) {
+	var res Upgrades
+	switch fs {
+	case SonicFeatures:
+		res = GetSonicUpgrades()
+	case AllegroFeatures:
+		res = GetAllegroUpgrades()
+	default:
+		return res, fmt.Errorf("unknown feature set: %v", fs)
+	}
+	return res, nil
+}
+
+func (fs FeatureSet) String() string {
+	switch fs {
+	case SonicFeatures:
+		return "sonic"
+	case AllegroFeatures:
+		return "allegro"
+	default:
+		return "unknown"
+	}
+}
+
+func GetSonicUpgrades() Upgrades {
+	return Upgrades{
+		Berlin:  true,
+		London:  true,
+		Llr:     false,
+		Sonic:   true,
+		Allegro: false,
+	}
+}
+
+func GetAllegroUpgrades() Upgrades {
+	return Upgrades{
+		Berlin:  true,
+		London:  true,
+		Llr:     false,
+		Sonic:   true,
+		Allegro: true,
+	}
+}
```

Comment:
```suggestion
func GetAllegroUpgrades() Upgrades {
    upgrades := GetSonicUpgrades()
    upgrades.Allegro := true
	return upgrades
}
```
wdyt?  makes more explicit the change from sonic to Allegro, less error prone to set a "previous" flag wrong.

Reply:
https://github.com/0xsoniclabs/sonic/pull/92/commits/ac1ae6eb4e86e450e4bdf61363a1a2fccedc2a22